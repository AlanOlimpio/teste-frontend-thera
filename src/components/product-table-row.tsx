"use client";

import { useRouter } from "next/navigation";
import { TableCell, TableRow } from "@/components/ui/table";

import { ProductInterfaceProps } from "@/interfaces/product";

import Image from "next/image";
import { categorysMap } from "@/utils/categorysMap";
import { priceFormatter } from "@/utils/formatter";

export function ProductTableRow({
  product,
}: {
  product: ProductInterfaceProps;
}) {
  const router = useRouter();

  return (
    <TableRow
      onClick={() => router.push(`/product/${product.id}`)}
      className="cursor-pointer hover:bg-gray-100 dark:hover:bg-gray-800"
    >
      <TableCell className="font-mono text-xs font-medium min-w-[90px]">
        {product.imageUrl ? (
          <Image
            width={80}
            height={80}
            src={product.imageUrl}
            alt={product.name}
            className="h-20 w-20 object-cover  rounded-md"
          />
        ) : (
          <div className="w-20 p-2  h-20 bg-gray-200 dark:bg-gray-700 rounded-md flex items-center justify-center">
            <span className="text-gray-500 dark:text-gray-400">Sem imagem</span>
          </div>
        )}
      </TableCell>
      <TableCell className="font-medium">{product.name}</TableCell>
      <TableCell className="font-medium">
        {categorysMap.map((category) => {
          if (category.value === product.category) {
            return category.label;
          }
        })}
      </TableCell>
      <TableCell>{priceFormatter.format(product.price)}</TableCell>
      <TableCell className="font-medium">
        {product.description?.substring(0, 50) || "Sem descrição"}
      </TableCell>
    </TableRow>
  );
}
