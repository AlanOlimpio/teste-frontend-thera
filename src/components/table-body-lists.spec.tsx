import { render, screen } from "@testing-library/react";
import { ProductTableRow } from "./product-table-row";

const productListMock = [
  {
    id: "6931037f-1ce4-4805-aefb-a6703cf99733",
    name: "Notebook Acer Nitro V15",
    description: "Processador: Intel Core i5-13420H (10 núcleos - 6P+4E)",
    category: "10",
    price: 4640,
    created_at: "2025-07-14T14:36:11.435Z",
    imageUrl: "/uploads/51Wv-tEUn6L._AC_SX679_.jpg",
    userId: null,
  },
];

jest.mock("next/navigation", () => ({
  useRouter: () => ({
    push: jest.fn(),
    replace: jest.fn(),
    refresh: jest.fn(),
    back: jest.fn(),
    forward: jest.fn(),
    prefetch: jest.fn(),
  }),
}));

describe("<ProductTableRow/>", () => {
  it("should display an item in the product list", () => {
    const { container } = render(
      <table>
        <tbody>
          <ProductTableRow
            product={productListMock[0]}
            key={`identifier${productListMock[0].id}`}
          />
        </tbody>
      </table>
    );

    const productName = screen.getByText("Notebook Acer Nitro V15");
    const productCategory = screen.getByText("Eletrodomésticos");
    const productPrice = screen.getByText("R$ 4.640,00");
    const productDescription = screen.getByText(
      "Processador: Intel Core i5-13420H (10 núcleos - 6P"
    );
    const productImage = screen.getByAltText(
      "Notebook Acer Nitro V15"
    ) as HTMLImageElement;

    expect(productName).toBeInTheDocument();
    expect(productCategory).toBeInTheDocument();
    expect(productPrice).toBeInTheDocument();
    expect(productDescription).toBeInTheDocument();
    expect(productImage.src).toContain(
      "/_next/image?url=%2Fuploads%2F51Wv-tEUn6L._AC_SX679_.jpg&w=256&q=75"
    );
    expect(container).toMatchSnapshot();
  });
});
