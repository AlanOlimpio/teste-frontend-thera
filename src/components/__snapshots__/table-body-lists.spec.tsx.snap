// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<ProductTableRow/> should display an item in the product list 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="border-b transition-colors data-[state=selected]:bg-muted cursor-pointer hover:bg-gray-100 dark:hover:bg-gray-800"
      >
        <td
          class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] font-mono text-xs font-medium min-w-[90px]"
        >
          <img
            alt="Notebook Acer Nitro V15"
            class="h-20 w-20 object-cover  rounded-md"
            data-nimg="1"
            decoding="async"
            height="80"
            loading="lazy"
            src="/_next/image?url=%2Fuploads%2F51Wv-tEUn6L._AC_SX679_.jpg&w=256&q=75"
            srcset="/_next/image?url=%2Fuploads%2F51Wv-tEUn6L._AC_SX679_.jpg&w=96&q=75 1x, /_next/image?url=%2Fuploads%2F51Wv-tEUn6L._AC_SX679_.jpg&w=256&q=75 2x"
            style="color: transparent;"
            width="80"
          />
        </td>
        <td
          class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] font-medium"
        >
          Notebook Acer Nitro V15
        </td>
        <td
          class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] font-medium"
        >
          Eletrodomésticos
        </td>
        <td
          class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]"
        >
          R$ 4.640,00
        </td>
        <td
          class="p-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px] font-medium"
        >
          Processador: Intel Core i5-13420H (10 núcleos - 6P
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
